lib/routes/tsdm39/bd.ts

```diff
@@ -0,0 +1,100 @@
+import type { DataItem, Route } from '@/types';
+import { config } from '@/config';
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import { parseDate } from '@/utils/parse-date';
+
+// type id => display name
+type Mapping = Record<string, string>;
+
+const TYPE: Mapping = {
+    '403': '720P',
+    '404': '1080P',
+    '405': 'BDMV',
+    '4130': '4K',
+    '5815': 'AV1',
+};
+
+// render into MD table
+const mkTable = (mapping: Mapping): string => {
+    const heading: string[] = [],
+        separator: string[] = [],
+        body: string[] = [];
+
+    for (const key in mapping) {
+        heading.push(mapping[key]);
+        separator.push(':--:');
+        body.push(key);
+    }
+
+    return [heading.join(' | '), separator.join(' | '), body.join(' | ')].map((s) => `| ${s} |`).join('\n');
+};
+
+const handler: Route['handler'] = async (ctx) => {
+    const { type } = ctx.req.param();
+
+    const cookie = config.tsdm39.cookie;
+
+    const item = (await cache.tryGet(`tsdm39:bd:${type ?? 'all'}`, async () => {
```

Comment:
Don't cache `route.handler.item` since RSSHub will cache it anyways.
Doing so will prevent the feed getting updated once it's cached.

Reply:
RSSHub will cache `route.handler.item` as one thing.

On the first request of a request path:
https://github.com/DIYgod/RSSHub/blob/88486f71d992bfa5c1147136eee40e54065c04fa/lib/registry.ts#L93 https://github.com/DIYgod/RSSHub/blob/88486f71d992bfa5c1147136eee40e54065c04fa/lib/middleware/cache.ts#L63-L69

On following requests of the same request path (within `config.cache.routeExpire`):
https://github.com/DIYgod/RSSHub/blob/88486f71d992bfa5c1147136eee40e54065c04fa/lib/middleware/cache.ts#L43-L51